src/puppetlabs/puppetdb/http/query.clj

```diff
@@ -56,6 +54,36 @@
                    ["=" ["node" "active"] true]]
                   req))
 
+(defn add-extract [projections query]
+  (cm/match [query]
+            [["extract" columns expr :guard nil?]]
```

Comment:
I think this needs to be (expr :guard nil?). I'm also not sure what it does -- probably needs test coverage, I think it must never be getting hit right now because :guard is just another keyword.